### PR TITLE
CI: Skip self-tests that will fail in PRs/without secrets anyway

### PR DESCRIPTION
**Pull request by rugk**
Was too lazy to look up the exact syntax, so ChatGPT-powered: https://chatgpt.com/share/4416cdb2-d192-480a-8aee-19317c5eea7b

The first commit shows one approach, the other one was a little more complex, but possibly better?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,7 +77,7 @@ jobs:
           channel: unlisted
           apiUrlPrefix: "https://addons.thunderbird.net/api/v3"
           apiKey: ${{ secrets.ATN_SIGN_KEY }}
-          apiSecret: ${{ secrets.ATN_SIGN_SECRET }}
+          apiSecret: ${{ secrets }}
```
Comment (rugk) [inline .github/workflows/ci.yml:80]:
```suggestion
          apiSecret: ${{ secrets.ATN_SIGN_SECRET }}
```

revert grr

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -80,7 +86,8 @@ jobs:
           path: ${{ steps.web-ext-build.outputs.target }}
 
       - name: "Upload AMO signed"
+        if: env.HAS_AMO_SIGN_KEY == 'true'
         uses: actions/upload-artifact@v4
         with:
           name: amo-signed.xpi
-          path: ${{ steps.web-ext-sign-amo.outputs.target }}
+          path: ${{ steps.web-ext-sign-amo.outputs.target }}
```
Comment (kewisch) [inline .github/workflows/ci.yml:93]: Newline at end of file please :)

**Comment by kewisch** [comment]
This looks great and should be fine. If there is a major PR with changes to how uploading works (like upcoming deps updates, ugh) I'll need to run it on my end, but at least this helps with small fixes.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rugk) [reply .github/workflows/ci.yml:93]:
```suggestion
          path: ${{ steps.web-ext-sign-amo.outputs.target }}
```

**Comment by rugk** [comment]
> If there is a major PR with changes to how uploading works (like upcoming deps updates, ugh) I'll need to run it on my end, but at least this helps with small fixes.

Well… I am not sure, did the CI work before for your own PRs? Because if, so it should continue to work as the latest version checks for the secrets to be set. If they are, it should run everything. If they are not, there is no point in running them, it will fail anyway…

**Comment by kewisch** [comment]
With "run on my end" I mean create the PR myself so it can run with secrets. Your changes should continue to work since my pull requests will include secrets.

**Pull request merged**